src/parquet/schema.h

```diff
@@ -264,8 +264,11 @@ class PARQUET_EXPORT GroupNode : public Node {
   bool Equals(const Node* other) const override;
 
   NodePtr field(int i) const { return fields_[i]; }
+  // Get the index of a field by its name, or negative value if not found
+  // If several fields share the same name, the smallest index is returned
```

Comment:
Couple of questions:

* I see [this language regarding the iteration order](http://en.cppreference.com/w/cpp/container/unordered_multimap) of the values for a particular key in the multimap:

> every group of elements whose keys compare equivalent (compare equal with key_eq() as the comparator) forms a contiguous subrange in the iteration order

Does the `iteration order` here mean that the values are iterated over in the order in which they were inserted?

* Why did you choose to return the first one instead of returning `-1` (or maybe `-2`) for the `std::string` overload? Do we not want to provide a way to indicate that column indexes and column names are not 1:1 in the C++ API? Maybe that already exists.

Reply:
1) That's a good point. The fact that the container is unordered means it isn't guaranteed to retain insertion order, even for values which map to the same key (I would expect a straightforward implementation to maintain that order, though). I should probably remove the sentence above.

2) Because doing otherwise seems like it could break compatibility. Not sure how strongly you feel about it. The `std::string` overloads aren't used anymore in the parquet-cpp codebase, AFAICT.

Reply:
> it could break compatibility

True, though IIUC wouldn't this potentially segfault if you tried to use the result to index into something?

Reply:
Why? We're still returning a valid index.

Reply:
As a side note, if @wesm wants to include this in the release, we can further API improvements to a later PR.

Reply:
I meant that before your change, the index returned was always valid?

Reply:
Yes, it was, it just wasn't necessarily the one expected by the caller according to its semantics.

Reply:
Right, makes sense.